app/Http/Controllers/UserController.php

```diff
@@ -2,10 +2,90 @@
 
 namespace App\Http\Controllers;
 
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+
 class UserController extends Controller
 {
     public function index()
     {
-        return view('auth.user');
+        $user = Auth::user();
+
+        return view('auth.user', ['user' => $user]);
+    }
+
+    /**
+     * Create a new controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->middleware('auth');
+    }
+
+    public function updateEmail(Request $request)
+    {
+        $user = Auth::user();
+
+        $validator = Validator::make($request->all(), [
+            'email' => 'required|email|max:225|unique:users',
+        ]);
+
+        if ($validator->fails()) {
+            return redirect()->back()
+                ->withErrors($validator)
+                ->withInput();
+        }
+        $user->update([
+            'email' => $request->email,
+        ]);
+
+        return redirect()->back();
+    }
+
+    public function updatePhone(Request $request)
+    {
+        $user = Auth::user();
+
+        $validator = Validator::make($request->all(), [
+            'phone_number' => 'required|string|min:16|max:18',
+        ]);
+
+        if ($validator->fails()) {
+            return redirect()->back()
+                ->withErrors($validator)
+                ->withInput();
+        }
+
+        $user->personalInformation->update([
+            'phone_number' => $request->phone_number,
+        ]);
+        //if user doesn`t have personalInformation TODO
```

Comment:
How about simply creating instead of updating for those without personalInformation? That would be a simple solution that won't give an error.

Reply:
Or you could simply hide the fields for these users, might be even better.

Reply:
> How about simply creating instead of updating for those without personalInformation? That would be a simple solution that won't give an error.

Yes, but we don't have all the information for that. I will try to hide it.

Reply:
You can add a member function to the user model:
`public function hasPersonalInformation(){
  return $this-> personalInformation === null;
}`
Haven't tested.

Reply:
Yes, but quite the opposite :)

Reply:
Yeah, but you can negate it :)